ivy/utils/backend/handler.py

```diff
@@ -236,9 +236,24 @@ def _handle_backend_specific_vars(target, backend):
         target.set_global_attr("RNG", target.functional.backends.jax.random.RNG)
 
 
-def convert_from_source_backend_to_numpy(variable_ids, numpy_objs, devices):
-    # Dynamic Backend
-    from ivy.functional.ivy.gradients import _is_variable, _variable_data
+def _data_to_new_backend(x, previous_backend):
```

Comment:
I'm not sure if I'm testing the wrong example, but if I run the following,
```
import ivy

ivy.set_backend("torch")
x = ivy.array([1., 2., 3.])
ivy.set_backend("tensorflow")
print(ivy.utils.backend.handler._data_to_new_backend(x, ivy.functional.backends.torch))
```
It seems to throw an error `ivy.utils.exceptions.IvyBackendException: tensorflow: dev: 'torch.device' object is not subscriptable`. Probably we could separate out the conversion to and from dlpack in the way we were doing it for conversion to and from numpy in 2 different steps, what do you think?